docs/reference-guide/backup.md

```diff
@@ -53,6 +53,19 @@ Alternatively, you can make one directly by running the following query:
 CREATE SNAPSHOT;
 ```
 
+#### Batches in snapshots
+
+The vertices and edges are divided into groups, called batches. When Memgraph
+starts to recover a snapshot it might read the data and build the indices on
+multiple threads. The batches serve as a unit of parallelization: each thread
+will get one batch at a time to recover until there are no unhandled batches.
+This means the same batch size might not be suitable for every dataset. A
+smaller dataset might require smaller batch size to utilize a multi-threaded
+processor, while bigger datasets might use bigger batches to minimize the
+synchronization between the worker threads. Therefore the size of batches and
+the number of threads used are configurable similarly to other durability
+related settings.
```

Comment:
```suggestion
When saved within a snapshot, vertices and edges are divided into groups called batches. When Memgraph
starts to recover a snapshot, it might read the data and build the indices on
multiple threads. The batches serve as a parallelization unit: each thread
will recover one batch at a time until there are no unhandled batches.

This means the same batch size might not be suitable for every dataset. A
smaller dataset might require a smaller batch size to utilize a multi-threaded
processor, while bigger datasets might use bigger batches to minimize the
synchronization between the worker threads. Therefore the size of batches and
the number of used threads [are configurable](/reference-guide/configuration#storage) similarly to other durability
related settings.
```

the link might cause an error, i did not check it :D I'll fix it if it does

Reply:
i think i would incorporate this new information in the existing paragraphs... so this is actually my suggestion:

### Snapshots
Snapshots provide a faster way to restore the states of your database. Memgraph
periodically takes snapshots during runtime. When a snapshot creation is triggered, the
entire data storage is written to the drive. Nodes and relationships are divided into groups called batches. 

On startup, the database state is recovered from the most recent snapshot file. Memgraph
can read the data and build the indices on
multiple threads, using batches as a parallelization unit: each thread
will recover one batch at a time until there are no unhandled batches.

This means the same batch size might not be suitable for every dataset. A
smaller dataset might require a smaller batch size to utilize a multi-threaded
processor, while bigger datasets might use bigger batches to minimize the
synchronization between the worker threads. Therefore the size of batches and
the number of used threads [are configurable](/memgraph/reference-guide/configuration#storage) similarly to other durability
related settings.

The timestamp of the snapshot is compared with the latest update recorded in
the WAL file and, if the snapshot is less recent, the state of the DB will be
recovered using the WAL file.

Memgraph has snapshot creation enabled by default. You can configure the exact
snapshot creation behavior by defining the relevant flags
([reference](/docs/memgraph/reference-guide/configuration#storage)).
Alternatively, you can make one directly by running the following query:
```opencypher
CREATE SNAPSHOT;
```

Reply:
> the link might cause an error, i did not check it :D I'll fix it if it does

I built it locally with `npm run build` and I don't see any error or warnings. I don't know if that what you meant or not, but seems working.